docs/SetupTestEnvironment.md

```diff
@@ -0,0 +1,38 @@
+# Setup Test Environment
+
+## Create a Service Principal
+
+- Create a new Service Principal on Azure using [azure cli](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?toc=%2Fen-us%2Fazure%2Fazure-resource-manager%2Ftoc.json&bc=%2Fen-us%2Fazure%2Fbread%2Ftoc.json&view=azure-cli-latest) or [azure portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal?view=azure-cli-latest)
+- Assign the new Service Principal with [Contributor Role]
+(https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals#assign-role-to-service-principal) so it can access resources on your subscription. This Service Principal will be assigned to Kubernetes to enable it to create and manage its own resources, as well as push build artifacts into Azure Container Registry.
+
+## Create an Azure Container Registry
+
+- Create a new [Azure Container Registry](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal#create-a-container-registry)
+
+## Create a Kubernetes Cluster
+
+- Create a new resource group on your subscription
+- Add a new resource and search for "Kubernetes Cluster"
+
+![Kubernetes Cluster](images/k8s-find.jpg)
+
+- Define the resource group, cluster name and DNS
+
+![Kubernetes Cluster](images/k8s-create.jpg)
+
+- Press "Next" and don't change any parameters in the authentication
+
+![Kubernetes Cluster](images/k8s-authenticate.jpg)
```

Comment:
It appears this image shows using an existing service principal. It would be better to let Azure create a new one. Can you update the image to look like this:

![image](https://user-images.githubusercontent.com/2847527/46769694-6f113600-cca1-11e8-8242-5480646cf63f.png)

Reply:
👍